src/data/nat/factorization/basic.lean

```diff
@@ -783,4 +783,173 @@ begin
     card_insert_eq_ite, IH, finset.mem_filter, mem_Ioc, not_le.2 (lt_add_one n)],
 end
 
+/-! ### Lemmas about factorizations of sums -/
```

Comment:
I understand that these results are about `nat.factorization`, but they're true in much more generality. Do we have the general results, probably using `multiplicity`? I am not sure we want to duplicate all results, but maybe we don't want to import `multiplicity`...